src/process.ml

```diff
@@ -329,11 +329,14 @@ let run_capture_gen ?dir ~env ?(purpose=Internal_job) fail_mode prog args ~f =
       Temp.destroy fn;
       x)
 
-let run_capture       = run_capture_gen ~f:Io.read_file
-let run_capture_lines = run_capture_gen ~f:Io.lines_of_file
+let run_capture =
+  run_capture_gen ~f:(fun p -> Io.read_file (Path.of_string p))
```

Comment:
What about changing `Temp.create` to return a `Path.t` instead?

Reply:
Yeah, there's a bunch of other places as well. Will push out separate commits.